agent/dns.go

```diff
@@ -793,32 +794,58 @@ func (d *DNSServer) doDispatch(network string, remoteAddr net.Addr, req, resp *d
 			if err != nil {
 				return invalid()
 			}
-
-			resp.Answer = append(resp.Answer, &dns.A{
-				Hdr: dns.RR_Header{
-					Name:   qName + d.domain,
-					Rrtype: dns.TypeA,
-					Class:  dns.ClassINET,
-					Ttl:    uint32(cfg.NodeTTL / time.Second),
-				},
-				A: ip,
-			})
+			//check if the query type is  A for IPv4 or ANY
+			if req.Question[0].Qtype != dns.TypeA && req.Question[0].Qtype != dns.TypeANY {
+				//resp.SetRcode(req, dns.RcodeSuccess)
+				resp.Extra = append(resp.Answer, &dns.A{
+					Hdr: dns.RR_Header{
+						Name:   qName + d.domain,
+						Rrtype: dns.TypeA,
+						Class:  dns.ClassINET,
+						Ttl:    uint32(cfg.NodeTTL / time.Second),
+					},
+					A: ip,
+				})
+			} else {
+				resp.Answer = append(resp.Answer, &dns.A{
+					Hdr: dns.RR_Header{
+						Name:   qName + d.domain,
+						Rrtype: dns.TypeA,
+						Class:  dns.ClassINET,
+						Ttl:    uint32(cfg.NodeTTL / time.Second),
+					},
+					A: ip,
+				})
```

Comment:
These two blocks are identical except for the field on the response where the value is set, is that right?

I think we could structure the code to make that more obvious:

```go

rr := &dns.A{...}
if req.Question[0].Qtype == dns.TypeA || req.Question[0].Qtype == dns.TypeANY {
    resp.Answer = append(resp.Answer, rr)
    return true
}

resp.Extra = append(resp.Extra, rr)
return true
```

(same below)